Migrate use to send_mail from EmailMultiAlternatives.

rohithasrk opened the pull request:
To Fix #3132 i.e., migrating use to `send_mail` from `EmailMultiAlternatives`. Made changes in the code which can be seen in the above commits. As @rishig suggested earlier, I tried building it upon PR #3137 , which now has been merged. Therefore there are commits by @rishig too. Hope it wouldn't be a problem.

Changes are made in files `zerver/management/commands/deliver_email.py` and `zerver/management/commands/send_password_reset_email.py` which were using `EmailMultiAlternatives`. Couldn't be made in `zerver/lib/notifications.py` as headers are being sent.

Please suggest changes. :)

smarx commented:
*Automated message from Dropbox CLA bot*

@rohithasrk, thanks for the pull request! It looks like you haven't yet signed the Dropbox CLA. Please [sign it here](https://opensource.dropbox.com/cla/).

smarx commented:
*Automated message from Dropbox CLA bot*
				
@rohithasrk, thanks for signing the CLA!

timabbott commented:
@rohithasrk thanks for working on this!  Can you rebase onto upstream/master (that should drop the duplicates of @rishig's commits)?  It looks like you also have some linter failures you'll want to fix as noted by Travis CI above.

Also, I notice you have a number of merge commits; Zulip doesn't use those.  Please use `git pull --rebase` when pulling upstream changes.  
Check out our [GitHub guide](http://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](http://zulip.readthedocs.io/en/latest/version-control.html) for more details.

rishig commented on zerver/management/commands/deliver_email.py:
```diff
@@ -36,6 +36,12 @@
 logger.setLevel(logging.DEBUG)
 logger.addHandler(file_handler)
 
+try:
```
Our use of `mailer` is deprecated, and can be removed (from here and anywhere else).

rohithasrk replied:
I see. Will be updating soon.

rohithasrk commented:
Was checking out the build errors shown by Travis CI, which can be seen in https://travis-ci.org/zulip/zulip/jobs/189918414. There are some errors due to long line in `docs/` folder which when I tried to correct couldn't because the text was in languages other than English, like Chinese, etc. Can anyone help me with it?

The error screenshot:
![trerror](https://cloud.githubusercontent.com/assets/13820931/21749236/dcf2c960-d5c0-11e6-82df-1bb4db276caf.png)

rohithasrk commented:
As @rishig suggested, I've removed all the code of `mailer` and fixed all the build errors. But I see that there are some more frontend-build errors. Can anyone please help me with it?

rishig commented:
cool! Where are you stuck? Do you know how to see where the errors are / test locally? In this case the errors are in `10-admin.js`, which you can test locally using `tools/test-js-with-casper 10-admin.js`.

rohithasrk commented:
Yeah @rishig. I pushed some commits which run well on my local system, pass all the tests. I don't understand why an error shows up for some change I feel I didn't do. Most of the errors in Travis CI are due to `setup-$TEST_SUITE` errors. Any idea where I have might gone wrong?

rishig commented:
Your latest changes seemed to have introduced real errors (e.g. in mypy), if I'm not mistaken. Could you clean up the commits (e.g. "Remove use of mailer .." could come at the beginning, rather than adding those blocks of code and them removing them), rebase on master, and push again? If you're getting casper errors that feel like they are not related to your change, it could be a flaky travis test, in which case we can merge it regardless of the error. (Casper has been really finicky for us :/.) 
Thank you! And sorry for unnecessary debugging :/.

rohithasrk commented:
@rishig : After some thorough checks, this seems to go fine with the tests. Anything else you think should be done?

rohithasrk commented:
@rishig : Made changes as you've suggested. @timabbott : Can you please review this? Thanks!

rohithasrk commented:
@timabbott : Changes done. Please review. Thanks :)

timabbott commented:
Merged, after rebasing the fixes in the third commit into the commits that introduced the errors (as is our standard practice).  Check out the Zulip commit version control guidelines for more details: https://zulip.readthedocs.io/en/latest/version-control.html

Thanks @rohithasrk!
